### PR TITLE
Add support for AMD SEV-SNP

### PR DESCRIPTION
**Pull request by dreemkiller**

### CODE REVIEW

File: crates/sev-veracruz-server/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+version = "0.1.0"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+anyhow = "1"
+bincode = { version = "1.2.1", default-features = false }
+env_logger = { version = "0.9.0" }
+log = "0.4.13"
+#nix = { version = "0.20.2" }
```
Comment (ShaleXIONG) [inline crates/sev-veracruz-server/Cargo.toml:15]: Unused comment.

File: crates/sev-veracruz-server/src/server.rs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            policy.proxy_attestation_server_url(),
+        )
+        .map_err(|err| {
+            eprintln!("Failed to start proxy attestation process. Error produced: {}.",
+                err,
+            );
+            err
+        })?;
+
+        let cid: u32 = 3; // TODO: Don't hard-code this
```
Comment (ShaleXIONG) [inline crates/sev-veracruz-server/src/server.rs:58]: TODO here? I do not see problem to hardcode, but maybe use all capital letters.

File: crates/veracruz-server/src/server.rs [hunk 3]
```diff
@@ -59,6 +59,7 @@ fn handle_veracruz_server_request<T: VeracruzServer + Sync + Send>(
         }
 
         // Response this request
+        // for x1 in output_data_option {
```
Comment (ShaleXIONG) [inline crates/veracruz-server/src/server.rs:62]: Unused comment.

File: docker/sev/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+         cd openssl-OpenSSL_1_1_1f/; \
+         CC="musl-gcc -fPIE -pie" ./Configure no-shared no-async --prefix=/musl --openssldir=/musl/ssl linux-x86_64; \
+         make depend; \
+         make -j$(nproc); \
+         make install;
+
+WORKDIR /
+RUN git clone --branch snp-latest https://github.com/AMDESE/AMDSEV.git
+WORKDIR /AMDSEV
+#RUN git checkout d80b494f712317493d464a55652698c4d1b7bb0f
```
Comment (ShaleXIONG) [inline docker/sev/Dockerfile:126]: Unused comments.

File: docker/sev/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+WORKDIR AMDSEV
+RUN git apply common.patch
+RUN git apply build.patch
+RUN git apply stable-commits.patch
+RUN ./build.sh --package qemu
+RUN ./build.sh --package kernel guest
+RUN ./build.sh --package ovmf
+
+WORKDIR /
+#RUN git clone https://github.com/virtee/sev-snp-measure.git
```
Comment (ShaleXIONG) [inline docker/sev/Dockerfile:143]: Unused comments.

File: crates/sev-veracruz-server/Cargo.toml (hunk 1 above)
Comment (dreemkiller) [reply crates/sev-veracruz-server/Cargo.toml:15]: Resolved.

File: crates/sev-veracruz-server/src/server.rs (hunk 2 above)
Comment (dreemkiller) [reply crates/sev-veracruz-server/src/server.rs:58]: resolved

File: crates/veracruz-server/src/server.rs (hunk 3 above)
Comment (dreemkiller) [reply crates/veracruz-server/src/server.rs:62]: resolved

File: docker/sev/Dockerfile (hunk 4 above)
Comment (dreemkiller) [reply docker/sev/Dockerfile:126]: resolved

File: docker/sev/Dockerfile (hunk 5 above)
Comment (dreemkiller) [reply docker/sev/Dockerfile:143]: resolved

**Review by ShaleXIONG — DISMISSED** [review]

**Comment by dreemkiller** [comment]
This is ready to merge if I can get someone to approve the `Cargo.lock` changes

**Comment by dreemkiller** [comment]
@ShaleXIONG or @gbryant-arm anyone wanna review the Cargo.lock changes?

**Review by ShaleXIONG — APPROVED** [review]

**Pull request merged**